### PR TITLE
[Build] separate docker file for cloud deployment

### PR DESCRIPTION
**Pull request by ZhiHanZ**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary

Summary about this PR
Introduced two docker images to separate meta and query instance.

## Changelog

- Build/Testing/CI

## Related Issues

Fixes https://github.com/datafuselabs/databend/issues/1970

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: docker/cloud/meta/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,4 @@
+FROM debian:bullseye
```
Comment (BohuTANG) [inline docker/cloud/meta/Dockerfile:1]: What's the `cloud` dir for?

Comment (ZhiHanZ) [reply docker/cloud/meta/Dockerfile:1]: Two put separated docker file

Comment (ZhiHanZ) [reply docker/cloud/meta/Dockerfile]: flattened the dir

**Pull request merged**